ledger-service/http-json/src/main/scala/com/digitalasset/http/dbbackend/ContractDao.scala

```diff
@@ -92,11 +152,24 @@ object ContractDao {
   def apply(
       cfg: JdbcConfig,
       tpIdCacheMaxEntries: Option[Long] = None,
+      diagnostic: Option[DiagnosticConfig] = None,
   )(implicit
       ec: ExecutionContext,
       metrics: HttpJsonApiMetrics,
   ): ContractDao = {
     val cs: ContextShift[IO] = IO.contextShift(ec)
+    def makeDatasourceConPoolAndExecutorService(
+        cfg: dbutils.JdbcConfig,
+        metricRegistry: Option[MetricRegistry] = None,
+    ) = {
+      // pool for connections awaiting database access
+      val es = Executors.newWorkStealingPool(cfg.poolSize)
```

Comment:
I think that for the diagnostics, we don't want a work stealing pool, but a single threaded one should be fine. There's not much value in concurrent diagnostics being dumped to the logs.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newSingleThreadExecutor--

Reply:
Agree. Can be done not big deal stealing diagnostic pool doesn't hurt

Reply:
I still think you're better off using the appropriate pool for the use case, which in this case would be the single threaded one.